File: src/main/java/org/vanautrui/languages/parsing/astnodes/terminal/CharConstNode.java
package org.vanautrui.languages.parsing.astnodes.terminal;

import org.vanautrui.languages.lexing.collections.TokenList;
import org.vanautrui.languages.lexing.tokens.CharConstantToken;
import org.vanautrui.languages.lexing.tokens.utils.IToken;
import org.vanautrui.languages.parsing.IASTNode;
import org.vanautrui.languages.parsing.astnodes.ITermNode;

public class CharConstNode implements IASTNode, ITermNode {

    public char content;

    public CharConstNode(TokenList tokens) throws Exception {

        IToken token = tokens.get(0);

        if (token instanceof CharConstantToken) {
            this.content = ((CharConstantToken) token).getContents().charAt(0);
            tokens.consume(1);
        } else {
            throw new Exception("could not read charConstant node");
        }

    }

    @Override
    public String toSourceCode() {
        return this.content+"";
    }
}

File: dragon-parser/src/main/parsing/CharConstNode.cpp
#include <iostream>
#include <vector>

//project headers
#include "CharConstNode.hpp"
#include "../commandline/TokenList.hpp"
#include "../commandline/TokenKeys.hpp"

using namespace std;

CharConstNode::CharConstNode(TokenList tokens, bool debug) {

	if(debug){
		cout << "CharConstNode(...)" << endl;
	}

	Token token = tokens.get(0);

	if (token.kind == CHARCONST) {
		this->content = token.value.at(0);
		tokens.consume(1);
	} else {
		throw ("could not read charConstant node");
	}

}

